soroban-env-host/src/host/error.rs

```diff
@@ -133,3 +133,9 @@ impl TryFrom<&HostError> for ScStatus {
         err.status.try_into()
     }
 }
+
+impl From<HostError> for std::io::Error {
+    fn from(e: HostError) -> Self {
+        std::io::Error::new(std::io::ErrorKind::Other, "host error")
```

Comment:
Instead of passing a string in here you should be able to pass HostError as the second argument. Then in the reverse conversion we should make it pull the HostError out.

Reply:
I fixed passing in the HostError. However, I was having had a hard time pulling it out in `metered_from_xdr` (mostly Rust-related). For now I just used `err_general` and added a `TODO` comment there. I might spend a bit more time on it later. But this PR should be ok to merge without it.

Reply:
Fixed it.